app/models/resource_type_validator.rb

```diff
@@ -0,0 +1,16 @@
+class ResourceTypeValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    invalid_resources = get_invalid_resources(value)
+    record.errors.add(attribute, "#{invalid_resources.join(', ')} is/are not a valid resource type(s)") if invalid_resources.any?
+  end
+
+  private
+
+    def get_invalid_resources(value)
+      authority = Qa::Authorities::Local::FileBasedAuthority.new('resource_types')
+      # the .present? check is to accomodate empty string/nil values since this
+      # isn't a required field, so ""/nil are valid entries
+      invalid_entries = value.map { |x| x if authority.search(x).blank? && x.present? }
```

Comment:
I don't think this is doing what you want. This will validate part of a two-word value. For example, the word "Part" will validate. `Hyrax::ResourceTypesService.select_options` is already a thing you could just check against allowed values, also the `authority` you are using has some methods that would work too. I think just skipping if blank/nil and then check if included in the valid options would be alot clearer logic.